File: 914 - X of a Kind in a Deck of Cards.py
from typing import Counter


class Solution:
    def hasGroupsSizeX(self, deck: list[int]) -> bool:
        count = Counter(deck)
        mincount = 9999
        for c in count.values():
            if mincount > c:
                mincount = c
        if mincount >= 2:
            for div in range(2, mincount + 1):
                res = True
                for c in count.values():
                    if c % div == 0:
                        continue
                    else:
                        res = False
                if res:
                    break
        else:
            res = False
        return res

File: 914 - X of a Kind in a Deck of Cards.js
/**
 * @param {number[]} deck
 * @return {boolean}
 */
var hasGroupsSizeX = function(deck) {
    let count = new Map();
    for (let card of deck) {
        if (count.has(card)) {
            count.set(card, count.get(card) + 1);
        } else {
            count.set(card, 1);
        }
    }
    function gcd(a, b) {
        return b == 0 ? a : gcd(b, a % b);
    }
    let g = 0;
    for (let co of count.values()) {
        g = gcd(co, g);
    }
    return g >= 2;
};